auth/unique-user-pool-domain.cjs

```diff
@@ -0,0 +1,12 @@
+const { createHash } = require('node:crypto')
```

Comment:
It turns out that the user pool domain needs to be unique, so I am following the same approach we took to generate a unique name for the bucket